### PR TITLE
Tests + CI

### PR DESCRIPTION
**Pull request by jacklee1792**

### CODE REVIEW

File: db/ddl.sql [hunk 1]
```diff
@@ -27,12 +29,27 @@ CREATE TABLE IF NOT EXISTS trades (
...
+CREATE TRIGGER ensure_valid_market_id_orders
+BEFORE INSERT ON orders
+BEGIN
+    SELECT RAISE(ABORT, 'Invalid market_id')
+    WHERE NEW.market_id NOT IN (SELECT id FROM markets);
+END;
+
+CREATE TRIGGER ensure_valid_market_id_trades
+BEFORE INSERT ON trades
+BEGIN
```
Comment (alicejjliu) [inline db/ddl.sql:52]: use FK key constraints instead

File: db/db.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 from .objects import Market, Order, Trade
 
 
 class Database:
-    def __init__(self, db: str):
-        self.db = db
-        self.conn, self.cursor = None, None
+    def __init__(self, conn: sqlite3.Connection):
+        self.conn = conn
+        self.conn.execute("PRAGMA foreign_keys = ON")
```
Comment (alicejjliu) [inline db/db.py:11]: also probably a good idea to turn on WAL mode too

Comment (alicejjliu) [reply db/db.py:11]: i think we can set this on database initialization though, no need for per-connection config

File: db/ddl.sql [hunk 3]
```diff
@@ -27,12 +29,27 @@ CREATE TABLE IF NOT EXISTS trades (
     price_cents INTEGER NOT NULL,
     quantity INTEGER NOT NULL,
     timestamp REAL DEFAULT CURRENT_TIMESTAMP,
-    FOREIGN KEY (market_id) REFERENCES markets (id)
+    FOREIGN KEY (market_id) REFERENCES markets (id) ON DELETE CASCADE
```
Comment (alicejjliu) [inline db/ddl.sql:32]: we shouldn't be deleting trades if markets get deleted, this makes PnL inaccurate

Comment (jacklee1792) [reply db/ddl.sql:32]: @alicejjliu You really shouldn't be deleting a market, my reasoning is that this is the equivalent of deleting a spreadsheet

But I think there should be an option to archive so it doesn't show up in /home

**Pull request merged**